app/controllers/topics_controller.rb

```diff
@@ -21,12 +22,24 @@ def create
       end
   end
 
-  private
+	def edit
+    @topic = Topic.find(params[:id])
+  end
+
+	def update
+    @topic = Topic.find(params[:id])
+    if @topic.update_attributes(topic_params)
+      flash[:success] = 'Topic updated'
+      redirect_to @topic
+    else
+      render 'edit'
+    end
+  end
+
+	private
```

Comment:
see https://github.com/bbatsov/ruby-style-guide
and find how to align private/protected methods